### PR TITLE
Deprecation class implemented to log when deprecated function is called.

### PR DESCRIPTION
**Pull request by alex-mtl**
Deprecation class implemented to log when deprecated function is called.

This implementation should allow to track if deprecated function is in use on production

First candidate to deprecate: translateContent()

### CODE REVIEW

File: library/Vanilla/Utility/Deprecation.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * This function outputs 1 message per function per request
+     *
+     * @return void
+     */
+    public static function log() {
+        $info = debug_backtrace()[1];
+        if (!key_exists($info['function'], self::$calls)) {
+            $fileName = str_replace(PATH_ROOT, '', $info['file']);
+            $message = 'Deprecated function '.$info['function'].' called from '.$fileName.' at line : '.$info['line'];
+            error_log($message, E_USER_ERROR);
```
Comment (tburry) [inline library/Vanilla/Utility/Deprecation.php]: If this also triggered a deprecate error then developers would see it too. I'm not sure if the debug bar can grab stuff from error_log.

Comment (tburry) [reply library/Vanilla/Utility/Deprecation.php]: You could also use `Vanilla\Logger::error(...)`.

Comment (alex-mtl) [reply library/Vanilla/Utility/Deprecation.php]: trigger_error added

File: library/core/functions.general.php [hunk 2]
```diff
@@ -3571,6 +3571,7 @@ function t($code, $default = false) {
      * @see Gdn::translate()
      */
     function translateContent($code, $default = false) {
```
Comment (initvector) [inline library/core/functions.general.php]: If we're going to deprecate a function, we should probably slap a [<span>@</span>deprecated tag](http://docs.phpdoc.org/references/phpdoc/tags/deprecated.html) in its doc block.

Comment (alex-mtl) [reply library/core/functions.general.php]: Done. Check here:
https://github.com/vanilla/vanilla/pull/7784/files

**Pull request merged**